includes/Core/Util/Tracking_Consent.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Util\Tracking_Consent
+ *
+ * @package   Google\Site_Kit\Core\Util
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Util;
+
+use Google\Site_Kit\Core\Permissions\Permissions;
+use Google\Site_Kit\Core\Storage\User_Setting;
+
+/**
+ * Class managing a user's anonymous usage tracking consent.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Tracking_Consent extends User_Setting {
+
+	/**
+	 * The user option name for this setting.
+	 *
+	 * @var string
+	 */
+	const OPTION = 'googlesitekit_tracking_optin';
+
+	/**
+	 * Registers the setting in WordPress.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function register() {
+		register_meta(
+			'user',
+			$this->user_options->get_meta_key( static::OPTION ),
+			array(
+				'type'         => 'boolean',
```

Comment:
We may want to use `get_type()` here instead, and we may also want to specify `get_default()` as `false`. Maybe we should even pass `sanitize_callback` as the return value of `get_sanitize_callback()` (without actually overriding that method).

This is all nit-picking, but these methods are protected, and it'd be kind of strange if `Tracking_Consent::get_type()` returned "string".

Reply:
Yeah good point about `get_type()` 👍